underfs/hdfs/src/main/java/alluxio/underfs/hdfs/acl/SupportedHdfsAclProvider.java

```diff
@@ -67,18 +71,70 @@ public AccessControlList getAcl(FileSystem hdfs, String path) throws IOException
     return acl;
   }
 
-  private AclEntryType getAclEntryType(AclEntry entry) throws IOException {
+  @Override
+  public void setAcl(FileSystem hdfs, String path, AccessControlList acl) throws IOException {
+    // convert AccessControlList into hdfsAcl
+    List<AclEntry> aclSpecs = new ArrayList<>();
+
+    for (alluxio.security.authorization.AclEntry entry : acl.getEntries()) {
+      AclEntry hdfsAclEntry = getHdfsAclEntry(entry);
+      aclSpecs.add(hdfsAclEntry);
+    }
+    // set hdfsAcl;
+    hdfs.setAcl(new Path(path), aclSpecs);
+    // set owner and owning group
+    hdfs.setOwner(new Path(path), acl.getOwningUser(), acl.getOwningGroup());
+  }
+
+  private AclEntry getHdfsAclEntry(alluxio.security.authorization.AclEntry entry)
+      throws IOException {
+    AclEntry.Builder builder = new AclEntry.Builder();
+    // Do not set name for unnamed entries
+    if (entry.getType() != alluxio.security.authorization.AclEntryType.OWNING_USER
+        && entry.getType() != alluxio.security.authorization.AclEntryType.OWNING_GROUP) {
+      builder.setName(entry.getSubject());
+    }
+
+    builder.setScope(entry.isDefault() ? AclEntryScope.DEFAULT : AclEntryScope.ACCESS);
+    builder.setType(getHdfsAclEntryType(entry));
+    FsAction permission = FsAction.getFsAction(entry.getActions().toCliString());
+    builder.setPermission(permission);
+    return builder.build();
+  }
+
+  private AclEntryType getHdfsAclEntryType(alluxio.security.authorization.AclEntry aclEntry)
+      throws IOException {
+    switch (aclEntry.getType()) {
+      case OWNING_USER:
+      case NAMED_USER:
+        return AclEntryType.USER;
+      case OWNING_GROUP:
+      case NAMED_GROUP:
+        return AclEntryType.GROUP;
+      case MASK:
+        return AclEntryType.MASK;
+      case OTHER:
+        return AclEntryType.OTHER;
+      default:
+        throw new IOException("Unknown Alluxio ACL entry type: " + aclEntry.getType());
+    }
+  }
+
+  private alluxio.security.authorization.AclEntryType getAclEntryType(AclEntry entry)
```

Comment:
Can you javadoc this that the param is an hdfs entry, and will return an Alluxio type?

Reply:
done